### ISSUE TITLE
Incompatible with newest web3

### ISSUE THREAD

**Issue by LoCorVin** [MAIAN-tool/MAIAN#16]
When i run mayan.py i get an import error:

File "maian.py", line 24, in <module>
    from web3 import Web3, KeepAliveRPCProvider, IPCProvider
ImportError: cannot import name 'KeepAliveRPCProvider'

my assumption is that the ''KeepAliveRPCProvider' was removed sometime. I compiled web3 from the current version 4.1 source. 

Do you use a different one, can you provide the version?

Regards,
Konrad

**Comment by LoCorVin** [comment]
One version that works is 3.4.4

**Comment by arun-munagala** [comment]
@LoCorVin using 3.4.4 gives me the following error

 File "/usr/local/lib/python3.5/dist-packages/web3/utils/crypto.py", line 26, in <module>
    assert sha3('') == 'c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470'
AssertionError.

any other workaround?

**Comment by LoCorVin** [comment]
No, until now i still have problems getting it to run. When analyzing a contract, a call to sha3 contracts.py expects a static signature, but web3 on 3.4.4 provides a signature expecting a argument self, so in theory we need a Web3 instance first.

**Comment by arun-munagala** [comment]
@LoCorVin Thanks for the update. even i have some problems. Tried using other web3 versions but had other dependency issues.

**Comment by LoCorVin** [comment]
If you ever get it running pleas tell me, maybe we can initiate a pull request with a Dockerfile that encapsulates all dependencies

**Comment by arun-munagala** [comment]
@LoCorVin Will surely let you know!

**Comment by nsumner** [comment]
After converting several divisions to integer divisions in execute_instruction.py, it works with python3.

I built the z3 python3 bindings and used these versions for dependencies:
 * web3 3.4.4
 * eth-abi 0.5.0
 * rlp 0.6.0

I have not done extensive testing, but the examples seem to work.
